resources/lib/telstra_auth.py

```diff
@@ -122,8 +122,14 @@ def get_token(username, password):
                 continue
             data = offer.get('productOfferingAttributes')
             ph_no = [x['value'] for x in data if x['name'] == 'ServiceId'][0]
-    except Exception:
-        raise TelstraAuthException('Unable to determine eligible services')
+            if 'ph_no' not in locals():
```

Comment:
No worries, I'll define as none first.
You're right, should get an index error if the list ends up empty but I think I've been looking at the wrong point of failure. We must be getting the unbound local errors from having an empty offers_list (although I'm not sure how this happens, no eligible services returns something else, but that's another issue I've just found). I'll fix it up later tonight.